karyon3-core/src/main/java/com/netflix/karyon/Karyon.java

```diff
@@ -333,84 +330,72 @@ public LifecycleInjector start() {
      * @param args - Runtime parameter (from main) injectable as {@literal @}Arguments String[]
      * @return the LifecycleInjector for this run
      */
-    public LifecycleInjector start(final String[] args) {
+    private LifecycleInjector start(Module module, final String[] args) {
         final Logger LOG = LoggerFactory.getLogger(Karyon.class);
         
-        for (Module module : modules) {
-            if (module instanceof KaryonModule) {
-                ((KaryonModule)module).configure(new KaryonBinder() {
-                    @Override
-                    public <T extends TypeLiteral<?>> void bindAutoBinder(
-                            Matcher<T> matcher, AutoBinder autoBinder) {
-                        Karyon.this.addAutoBinder(matcher, autoBinder);
-                    }
-                });
+        final KaryonFeatureSetImpl featureSet = new KaryonFeatureSetImpl(new IdentityHashMap<>(featureOverrides));
+        
+        if (featureSet.get(KaryonFeatures.DISCOVER_AUTO_BINDERS)) {
+            for (AutoBinder autoBinder : ServiceLoader.load(AutoBinder.class)) {
```

Comment:
If we're going to use ServiceLoader, why not just use it for the conditional bindings too and make them a little more friendly for users?